### PR TITLE
fix(lookup): use newValue for digest lookup, not compareValue

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Fixes a regression error where digest updates for images with versionCompatibility would be incorrect, e.g. using the digest for node:20.11.1 instead of node:20.11.1-alpine

## Context

Fixes #27584

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/process/lookup/index.ts [hunk 1]
```diff
@@ -442,14 +442,32 @@ export async function lookupUpdates(
...
+
     // Add digests if necessary
     if (supportsDigests(config.datasource)) {
       if (config.currentDigest) {
         if (!config.digestOneAndOnly || !res.updates.length) {
           // digest update
           res.updates.push({
             updateType: 'digest',
-            newValue: compareValue,
+            newValue: config.currentValue,
```
Comment (viceice) [inline lib/workers/repository/process/lookup/index.ts:470]: shouldn't this newValue?

File: lib/workers/repository/process/lookup/index.ts [hunk 2]
```diff
@@ -459,7 +477,7 @@ export async function lookupUpdates(
           res.updates.push({
             isPinDigest: true,
             updateType: 'pinDigest',
-            newValue: compareValue,
+            newValue: config.currentValue,
```
Comment (viceice) [inline lib/workers/repository/process/lookup/index.ts:480]: here too?

Comment (rarkins) [reply lib/workers/repository/process/lookup/index.ts:480]: newValue is the same as currentValue. There is no config.newValue

File: lib/workers/repository/process/lookup/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/workers/repository/process/lookup/index.ts:470]: There's no newValue here. The update is being created and it's same as currentValue

**Comment by rarkins** [comment]
@viceice this fix is important to get merged/released/deployed today

**Pull request merged**